python/pyspark/pandas/groupby.py

```diff
@@ -993,6 +993,115 @@ def nth(self, n: int) -> FrameLike:
 
         return self._prepare_return(DataFrame(internal))
 
+    def prod(self, numeric_only: Optional[bool] = True, min_count: int = 0) -> FrameLike:
+        """
+        Compute prod of groups.
+
+        .. versionadded:: 3.4.0
+
+        Parameters
+        ----------
+        numeric_only : bool, default False
+            Include only float, int, boolean columns. If None, will attempt to use
+            everything, then use only numeric data.
+
+        min_count: int, default 0
+            The required number of valid values to perform the operation.
+            If fewer than min_count non-NA values are present the result will be NA.
+
+        Returns
+        -------
+        Series or DataFrame
+            Computed prod of values within each group.
+
+        See Also
+        --------
+        pyspark.pandas.Series.groupby
+        pyspark.pandas.DataFrame.groupby
+
+        Examples
+        --------
+        >>> df = ps.DataFrame(
+        ...     {
+        ...         "A": [1, 1, 2, 1, 2],
+        ...         "B": [np.nan, 2, 3, 4, 5],
+        ...         "C": [1, 2, 1, 1, 2],
+        ...         "D": [True, False, True, False, True],
+        ...     }
+        ... )
+
+        Groupby one column and return the prod of the remaining columns in
+        each group.
+
+        >>> df.groupby('A').prod().sort_index()
+             B  C  D
+        A
+        1  8.0  2  0
+        2  15.0 2  1
+
+        >>> df.groupby('A').prod(min_count=3).sort_index()
+             B  C   D
+        A
+        1  NaN  2.0  0.0
+        2  NaN NaN  NaN
+        """
+
+        self._validate_agg_columns(numeric_only=numeric_only, function_name="prod")
+
+        groupkey_names = [SPARK_INDEX_NAME_FORMAT(i) for i in range(len(self._groupkeys))]
+        internal, agg_columns, sdf = self._prepare_reduce(
+            groupkey_names=groupkey_names,
+            accepted_spark_types=(NumericType, BooleanType),
+            bool_to_numeric=True,
+        )
+
+        psdf: DataFrame = DataFrame(internal)
+        if len(psdf._internal.column_labels) > 0:
+
+            stat_exprs = []
+            for label in psdf._internal.column_labels:
+                label_name = label[0]
+                tmp_count_column_name = verify_temp_column_name(
+                    sdf, "__tmp_%s_count_col__" % label_name
+                )
+                psser = psdf._psser_for(label)
+                column = psser._dtype_op.nan_to_null(psser).spark.column
+                data_type = psser.spark.data_type
+
+                if isinstance(data_type, IntegralType):
+                    stat_exprs.append(F.product(column).cast(data_type).alias(label_name))
+                else:
+                    stat_exprs.append(F.product(column).alias(label_name))
+
+                if min_count > 0:
+                    stat_exprs.append(F.count(column).alias(tmp_count_column_name))
+
+            sdf = sdf.groupby(*groupkey_names).agg(*stat_exprs)
```

Comment:
Can we simplify the logic like this?

```suggestion
            for label in psdf._internal.column_labels:
                psser = psdf._psser_for(label)
                scol = psser._dtype_op.nan_to_null(psser).spark.column
                prod_scol = F.product(scol)
                if min_count > 0:
                    prod_scol = F.when(F.count(scol) < min_count, F.lit(None)).otherwise(F.product(scol))
                else:
                    prod_scol = F.product(scol)
                
                stat_exprs.append(
                    prod_scol.alias(psser._internal.data_spark_column_names[0])
                )
            sdf = sdf.groupby(*groupkey_names).agg(*stat_exprs)
```